src/user_input.rs

```diff
@@ -1,444 +1,814 @@
+    fn from(input: KeyCode) -> Self {
+        InputKind::Keyboard(input)
+    }
+}
+
+impl From<MouseButton> for InputKind {
+    fn from(input: MouseButton) -> Self {
+        InputKind::Mouse(input)
+    }
+}
+
+/// A collection of [`Input`] structs, which can be used to update an [`InputMap`](crate::input_map::InputMap).
+///
+/// Each of these streams is optional; if a stream does not exist, it is treated as if it were entirely unpressed.
+///
+/// These are typically collected via a system from the [`World`](bevy::prelude::World) as resources.
+#[derive(Debug, Clone)]
+pub struct InputStreams<'a> {
+    /// An optional [`GamepadButton`] [`Input`] stream
+    pub gamepad_buttons: Option<&'a Input<GamepadButton>>,
+    /// An optional [`GamepadButton`] [`Axis`] stream
+    pub gamepad_button_axes: Option<&'a Axis<GamepadButton>>,
+    /// An optional [`GamepadAxis`] [`Axis`] stream
+    pub gamepad_axes: Option<&'a Axis<GamepadAxis>>,
+    /// An optional [`KeyCode`] [`Input`] stream
+    pub keyboard: Option<&'a Input<KeyCode>>,
+    /// An optional [`MouseButton`] [`Input`] stream
+    pub mouse: Option<&'a Input<MouseButton>>,
+    /// The [`Gamepad`] that this struct will detect inputs from
+    pub associated_gamepad: Option<Gamepad>,
+}
+
+// Constructors
+impl<'a> InputStreams<'a> {
+    /// Construct [`InputStreams`] with only a [`GamepadButton`] input stream
+    pub fn from_gamepad(
+        gamepad_button_stream: &'a Input<GamepadButton>,
+        gamepad_button_axis_stream: &'a Axis<GamepadButton>,
+        gamepad_axis_stream: &'a Axis<GamepadAxis>,
+        associated_gamepad: Gamepad,
+    ) -> Self {
+        Self {
+            gamepad_buttons: Some(gamepad_button_stream),
+            gamepad_button_axes: Some(gamepad_button_axis_stream),
+            gamepad_axes: Some(gamepad_axis_stream),
+            keyboard: None,
+            mouse: None,
+            associated_gamepad: Some(associated_gamepad),
+        }
+    }
+
+    /// Construct [`InputStreams`] with only a [`KeyCode`] input stream
+    pub fn from_keyboard(keyboard_input_stream: &'a Input<KeyCode>) -> Self {
+        Self {
+            gamepad_buttons: None,
+            gamepad_button_axes: None,
+            gamepad_axes: None,
+            keyboard: Some(keyboard_input_stream),
+            mouse: None,
+            associated_gamepad: None,
+        }
+    }
+
+    /// Construct [`InputStreams`] with only a [`GamepadButton`] input stream
+    pub fn from_mouse(mouse_input_stream: &'a Input<MouseButton>) -> Self {
+        Self {
+            gamepad_buttons: None,
+            gamepad_button_axes: None,
+            gamepad_axes: None,
+            keyboard: None,
+            mouse: Some(mouse_input_stream),
+            associated_gamepad: None,
+        }
+    }
+}
+
+// Input checking
+impl<'a> InputStreams<'a> {
+    /// Is the `input` matched by the [`InputStreams`]?
+    pub fn input_pressed(&self, input: &UserInput) -> bool {
```

Comment:
@alice-i-cecile Instead of "pressed", would it be clearer to (someday) use the term "active" or "activated"? This returns `true` for an axis if it is outside of the deadzone.

Reply:
I'm on board with that in a follow-up PR.

Reply:
See #157.